### PR TITLE
addstr asserts even with 0 to print, so break out early

### PR DESCRIPTION
**Pull request by lastquestion**
While investigating #118, I discovered a bug where we are causing `addstr` to throw a `curses` exception - at least on my mac osx, `addstr` excepts if you try to print outside the screen, even if the number of characters to print is 0.

also, added a test for super long file names and make sure they are truncated correctly.

### CODE REVIEW

File: src/__tests__/testScreen.py [hunk 1]
```diff
@@ -173,7 +182,7 @@ def assertEqualLines(self, testName, actualLines, expectedLines):
             self.assertEqual(
                 expectedLine,
                 actualLine,
-                'Lines did not match for test %s:\n\nExpected:%s\nActual:%s' % (
+                'Lines did not match for test %s:\n\nExpected:%s\nActual  :%s' % (
```
Comment (lastquestion) [inline src/__tests__/testScreen.py]: add some spaces here to line up the expected and actual for easy visual diffing.

File: src/__tests__/inputs/longFileNames.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+longfilenamereallylongfilenameitssuperlongareyoutiredoftryingtoreadthisyet
```
Comment (pcottle) [inline src/__tests__/inputs/longFileNames.txt]: haha

File: src/__tests__/testScreen.py [hunk 3]
```diff
@@ -83,6 +83,15 @@
     'input': 'gitDiffSomeExist.txt',
     'validateFileExists': True,
     'withAttributes': True,
+}, {
+    'name': 'longFileNames',
+    'input': 'longFileNames.txt',
+    'validateFileExists': False,
+    'withAttributes': False,
+    'screenConfig': {
```
Comment (pcottle) [inline src/__tests__/testScreen.py]: glad you found the screen config useful

File: src/__tests__/testScreen.py (hunk 1 above)
Comment (pcottle) [inline src/__tests__/testScreen.py]: ah great idea -- hadnt thought about the character differences between the two labels. maybe we should use tabs...

**Comment by pcottle** [comment]
I had a small merge conflict on the `Lines did not match for test %s:\n\nExpected:%s\nActual  :%` line from your other PR, so I just combined the two (visually lined them up and wrapped in quotes too)

**Comment by lastquestion** [comment]
Yeah, that other fix would have caused a merge conflict. thanks for fixing it up! :+1:

**Pull request merged**